network/network_linux.go

```diff
@@ -179,6 +211,41 @@ func (nm *networkManager) addBridgeRules(extIf *externalInterface, hostIf *net.I
 	return nil
 }
 
+func (nm *networkManager) addOVSRules(extIf *externalInterface, hostIf *net.Interface, bridgeName string) error {
+	primary := extIf.IPAddresses[0].IP.String()
+	mac := extIf.MacAddress.String()
+	macHex := strings.Replace(mac, ":", "", -1)
+
+	cmd := fmt.Sprintf("ovs-ofctl add-flow %s ip,nw_dst=%s,dl_dst=%s,priority=20,actions=normal", bridgeName, primary, mac)
+	_, err := common.ExecuteShellCommand(cmd)
+	if err != nil {
+		log.Printf("[net] Adding SNAT rule failed with error %v", err)
+		return err
+	}
+
+	log.Printf("[net] Get ovs port for interface %v.", hostIf.Name)
+	cmd = fmt.Sprintf("ovs-vsctl get Interface %s ofport", hostIf.Name)
+	ofport, err := common.ExecuteShellCommand(cmd)
+	if err != nil {
+		log.Printf("[net] Get ofport failed with error %v", err)
+		return err
+	}
+
+	ofport = strings.Trim(ofport, "\n")
+
+	// Add DNAT rule to forward ARP replies to container interfaces.
+	log.Printf("[net] Adding DNAT rule for ingress ARP traffic on interface %v.", hostIf.Name)
```

Comment:
I saw we have a few places using similar code blocks, can we reuse them?

Reply:
stuff like get ovs port appears in several places. But this is fine, as those are not big code blocks.